x-pack/plugin/ccr/qa/downgrade-to-basic-license/build.gradle

```diff
@@ -53,11 +50,11 @@ tasks.register("writeJavaPolicy") {
 
 tasks.register("follow-cluster", RestIntegTestTask) {
   dependsOn 'writeJavaPolicy', "leader-cluster"
-    useCluster testClusters."leader-cluster"
+    useCluster leaderCluster.get()
```

Comment:
Can we provide an overloaded version of `useCluster` on `TestClusterAware` that accepts a `Provider` so we can avoid all these calls to `get()`?

Reply:
could point. I'll give the testclusters api bit more attention

Reply:
I did some polishing on the api. Besides allow passing a task provider here I also added support for Provider in nonSystemProperties. this makes configurations based on clusters more straight forward by allowing using provider mapping: e.g.

```
tasks.register("someRestIntegTest", StandaloneRestIntegTestTask) {
    useCluster baseCluster
    nonInputProperties.systemProperty('tests.rest.cluster', baseCluster.map(c->c.allHttpSocketURI.join(",")))
}
```

Reply:
Nice!